Frontend/Frontend.Client/Pages/LoginPage.razor

```diff
@@ -77,7 +77,24 @@
             await WebSocketService.ConnectAsync($"ws://{userInfo.ip}:{userInfo.port}");
             await WebSocketService.SendMessageAsync("DO AUTH\r\nFROM " + userInfo.username);
 
-            NavigationManager.NavigateTo("main-page");
+            var authTimeout = TimeSpan.FromSeconds(10);
+            var startTime = DateTime.Now;
+
+            while (!WebSocketService.IsAuthenticated && DateTime.Now - startTime < authTimeout)
+            {
+                await Task.Delay(100);
+            }
+
+            isLoading = false;
+
+            if (WebSocketService.IsAuthenticated)
+            {
+                NavigationManager.NavigateTo("main-page");
+            }
+            else
+            {
+                errorMesages.Add("Failed to authenticate. Please check your credentials.");
```

Comment:
The server returns an error message with the REFUSE, why are we not using it here?